### PR TITLE
[TOOLS-2574] New sheet added to the FollowUp Report, containing the h…

### PR DESCRIPTION
**Pull request by emannuelalmeida**
…istory.

### CODE REVIEW

File: core/src/main/java/objective/taskboard/data/AuditSnapshotDto.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.javers.core.metamodel.object.CdoSnapshot;
+
+@SuppressWarnings("serial")
+public class AuditSnapshotDto implements Serializable {
+
+    private Map<String, Object> properties;
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/data/AuditSnapshotDto.java]: CRITICAL Make "properties" transient or serializable. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1948)

File: core/src/main/java/objective/taskboard/project/config/ProjectClusterHistoryService.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        this.auditService = auditService;
+        this.baseClusterService = baseClusterService;
+    }
+
+    public List<ProjectClusterHistoryItem> getProjectClusterHistory(ProjectFilterConfiguration project){
+        List<ProjectClusterHistoryItem> history = new ArrayList<>();
+
+        if (project.getBaseClusterId().isPresent()) {
+
+            SizingCluster baseCluster = baseClusterService.findById(project.getBaseClusterId().get());
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/project/config/ProjectClusterHistoryService.java]: MAJOR Call "Optional#isPresent()" before accessing the value. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS3655)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 2 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: followup-report/src/main/java/objective/taskboard/followup/FollowUpReportGenerator.java [hunk 3]
```diff
@@ -71,7 +75,7 @@ public FollowUpReportGenerator(SpreadsheetEditor editor, FieldMetadataService fi
         this.throughputKpiService = throughputKpiService;
     }
 
-    public Resource generate(FollowUpSnapshot snapshot, ZoneId timezone) 
+    public Resource generate(FollowUpSnapshot snapshot, ZoneId timezone, List<ProjectClusterHistoryItem> clusterHistory)
```
Comment (taksan) [inline followup-report/src/main/java/objective/taskboard/followup/FollowUpReportGenerator.java]: Add the following method before, so you won't need to change existing tests:
```
public Resource generate(FollowUpSnapshot snapshot, ZoneId timezone) {
    return generate(snapshot,  timezone, Collections.emptyList());
}
```

File: followup-report/src/test/java/objective/taskboard/followup/FollowUpReportGeneratorTest.java [hunk 4]
```diff
@@ -116,7 +121,7 @@ public void givenEmptyTransitions_whenGenerateFromJiraSheet_thenShouldNotGenerat
         FollowUpData followupData = new FollowUpData(fromJiraDS, getEmptyAnalyticsTransitionsDataSet(),
                 getEmptySyntheticTransitionsDataSet());
 
-        subject.generate(mockSnapshot(followupData), timezone);
+        subject.generate(mockSnapshot(followupData), timezone, clusterHistory);
```
Comment (taksan) [inline followup-report/src/test/java/objective/taskboard/followup/FollowUpReportGeneratorTest.java]: No need for this: `, clusterHistory`

File: core/src/main/java/objective/taskboard/project/config/ProjectClusterHistoryItem.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        str.append(cycle);
+
+        return str.toString();
+    }
+
+    public String getIssueType() {
+        return issueType;
+    }
+
+    public Long getDate() {
```
Comment (taksan) [inline core/src/main/java/objective/taskboard/project/config/ProjectClusterHistoryItem.java]: Replace by:
```
    public LocalDateTime getLocalDate(ZoneId tz) {
        if (date == null) return  null;
        return LocalDateTime.ofInstant(Instant.ofEpochMilli(date), tz);
    }
```

**Pull request merged**